File: src/main/java/com/kamegatze/learnjvm/model/db/roles/Roles.java
package com.kamegatze.learnjvm.model.db.roles;

import com.kamegatze.learnjvm.model.db.users.Users;
import jakarta.persistence.*;
import org.springframework.data.jpa.domain.AbstractPersistable;

import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "roles")
public class Roles extends AbstractPersistable<UUID> {
    @Column(name = "name")
    @Enumerated(EnumType.STRING)
    private ERoles name;

    @OneToMany(mappedBy = "role")
    private List<Users> users;

    public Roles() {
    }

    public Roles(UUID id, ERoles name) {
        this.setId(id);
        this.name = name;
    }

    public ERoles getName() {
        return name;
    }

    public void setName(ERoles name) {
        this.name = name;
    }

    public List<Users> getUsers() {
        return users;
    }

    public void setUsers(List<Users> users) {
        this.users = users;
    }
}

File: src/main/kotlin/com/kamegatze/learnjvm/model/db/roles/Roles.kt
package com.kamegatze.learnjvm.model.db.roles

import com.kamegatze.learnjvm.model.db.Entity
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table

@Table(value = "roles")
data class Roles(
    @field:Column(value = "name")
    var name: String) : Entity() {
    constructor() : this( "")
}
